src/components/disclosure/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC, useState } from 'react'
+import classNames from 'classnames'
+import { IconChevronRight24 } from '@hashicorp/flight-icons/svg-react/chevron-right-24'
+import Text from 'components/text'
+import { DisclosureProps } from './types'
+import s from './disclosure.module.css'
+
+const Disclosure: FC<DisclosureProps> = ({
+  children,
+  description,
+  id,
+  open = false,
+  title,
+}) => {
+  const [isOpen, setIsOpen] = useState(open)
+  const containerId = `disclosure-${id}`
+  const contentContainerId = `${containerId}-content`
+  const rootClassnames = classNames(
+    s.root,
+    s[`root-${isOpen ? 'expanded' : 'collapsed'}`]
+  )
+
+  return (
+    <div className={rootClassnames} id={containerId}>
+      <button
+        aria-controls={contentContainerId}
+        aria-expanded={isOpen}
+        className={s.button}
+        onClick={() => setIsOpen(!isOpen)}
```

Comment:
nit, consider using a setter function here to avoid any potential stale values:
```suggestion
        onClick={() => setIsOpen(currentIsOpen => !currentIsOpen)}
```

Reply:
Good call!